prymer/thermo.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import dataclass
+from dataclasses import field
+from typing import ClassVar
+
+from primer3.thermoanalysis import ThermoResult  # type: ignore
+from primer3.thermoanalysis import _ThermoAnalysis
+
+
+@dataclass(frozen=True, kw_only=True)
+class Thermo:
+    """
+    Class for performing thermodynamic calculations.  Available calculations include:
+
+      1. melting temperature (Tm) for short and long sequences
+      2. hairpin / secondary structure Tm for single sequences
+      3. homodimer Tm - the Tm of a duplex formed from two copies of the same sequence
+      4. heterodimer Tm - the Tm of a duplex formed from two different sequences
+      5. 3' anchored heterodimer Tm - the heterodimer Tm when annealing of the 3' end is prioritized
+
+    The `tm` method can be used for sequences of any length.  For sequences up to `max_nn_length`
+    (default value 60 bases) the nearest neighborhood Tm calculation is used.  For sequences above
+    the `max_nn_length` a more generic long sequence Tm calculation is used.
+
+    Many attributes affect the calculation of Tms and these are set to the default values used by
+    primer3 and, where applicable, ntthal.
+
+    Class level constants are provided for Tm calculation method names and salt correction method
+    names, and it is strongly suggested to use these.
+
+    Attributes:
+        mv_conc_mm: concentration of monovalent cations in mM
+        dv_conc_mm: concentration of divalent cations in mM
+        dntp_conc_mm: concentration of deoxynycleotide triphosphate in mM
+        dna_conc_nm: concentration of DNA strands in nM
+        dmso_conc_pct: Concentration of DMSO (%)
+        dmso_correction_factor: DMSO correction factor
+        formamide_conc_mol: Concentration of formamide (mol/l)
+        annealing_temp_c: Actual annealing temperature of the PCR reaction in Celsius
+        temp_c: Simulation temperature for structure prediction in Celsius
+        max_nn_length: Maximum length in bases for nearest-neighbor calcs
+        max_loop: maximum size of loops in bases when predicting structures
+        tm_method: Tm calculation method
+        salt_corrections_method: Salt correction method
+    """
+
+    TM_METHOD_BRESLAUER: ClassVar[str] = "breslauer"
+    TM_METHOD_SANTALUCIA: ClassVar[str] = "santalucia"
+
+    SALT_CORRECTION_METHOD_SCHILDKRAUT: ClassVar[str] = "schildkraut"
+    SALT_CORRECTION_METHOD_OWCZARZY: ClassVar[str] = "owczarzy"
+    SALT_CORRECTION_METHOD_SANTALUCIA: ClassVar[str] = "santalucia"
```

Comment:
I think you could define the enums at the class level, to keep the namespace from being polluted.

Also consider using [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum).  We may want to use [`LowercaseStrEnum `](https://strenum.readthedocs.io/en/latest/api_ref.html#strenum.LowercaseStrEnum) from `strenum` as `StrEnum` isn't in the standard library until 3.11.

```suggestion
    @enum.unique
    class TmMethod(Enum):
        BRESLAUER = "breslauer"
        SANTALUCIA = "santalucia"
    
    @enum.unique
    class SaltCorrectionMethod(Enum):
        SCHILDKRAUT = "schildkraut"
        OWCZARZY = "owczarzy"
        SANTALUCIA = "santalucia"
```

Reply:
I'm going to defer this for now.  I hear you .. but actually the way these same things get provided to primer3 during design is as _integers_ (i.e. 0, 1, or 2).  So I want to think about where these belong and how best to manage them when I circle back and add them there.